### PR TITLE
Changes needed to make moab properly export hdf5 properly.

### PR DESCRIPTION
**Pull request by robertmaynard**
Changes that are needed for smtk version of moab to properly report what version of hdf5 it is built against.

These changes have been sent to @mathstuf so that they can be merged into the moab branch for the superbuild ( and maybe for upstream )

This issue was detected when creating https://gitlab.kitware.com/cmb/cmb-superbuild/merge_requests/83

### CODE REVIEW

**Comment by mathstuf** [comment]
https://bitbucket.org/fathomteam/moab/pull-requests/179/make-moab-properly-export-the-hdf5/diff

File: thirdparty/moab/MOABConfig.new.cmake.in [hunk 1]
```diff
@@ -50,14 +50,8 @@ set(MOAB_HAS_FBIGEOM "@ENABLE_IGEOM@")
 set(MOAB_HAS_IMESH "@ENABLE_IMESH@")
 set(MOAB_HAS_MPI "@MOAB_HAS_MPI@")
 
-if(MOAB_USE_HDF)
-  if(EXISTS "@HDF5_DIR@/share/cmake/hdf5/hdf5-config.cmake")
-    include(@HDF5_DIR@/share/cmake/hdf5/hdf5-config.cmake)
-  endif()
-endif()
```
Comment (mathstuf) [inline thirdparty/moab/MOABConfig.new.cmake.in]: Upstream wants to know why this was removed. My guess is that HDF5's config file isn't always at this location?

Comment (robertmaynard) [inline thirdparty/moab/MOABConfig.new.cmake.in]: I was just ripping stuff out that I didn't need.

I think the proper fix is to change it to
```
if(MOAB_USE_HDF)
  if(EXISTS "@HDF5_DIR@/hdf5-config.cmake")
    include(@HDF5_DIR@/hdf5-config.cmake)
  endif()
endif()
```
As at least for me HDF5_DIR points to ```<install_prefix>/share/cmake/hdf5/``` already.

I will update this branch to do exactly this.

Comment (mathstuf) [inline thirdparty/moab/MOABConfig.new.cmake.in]: Well, comments upstream note that the `hdf5-config.cmake` file doesn't seem to even exist in their installs. So if it isn't needed, I suspect we can just rip it out.

File: thirdparty/moab/MOABConfig.new.cmake.in [hunk 2]
```diff
@@ -51,13 +51,14 @@ set(MOAB_HAS_IMESH "@ENABLE_IMESH@")
 set(MOAB_HAS_MPI "@MOAB_HAS_MPI@")
 
 if(MOAB_USE_HDF)
-  if(EXISTS "@HDF5_DIR@/share/cmake/hdf5/hdf5-config.cmake")
-    include(@HDF5_DIR@/share/cmake/hdf5/hdf5-config.cmake)
+  if(EXISTS "@HDF5_DIR@/hdf5-config.cmake")
+    include(@HDF5_DIR@/hdf5-config.cmake)
```
Comment (mathstuf) [inline thirdparty/moab/MOABConfig.new.cmake.in]: Yeah…not sure why this isn't just `find_package(HDF5 QUIET)`, but meh.

**Pull request merged**